wherehows-frontend/app/controllers/api/v2/Dataset.java

```diff
@@ -246,6 +250,46 @@ private static AclDao initAclDao() {
     return Promise.promise(() -> ok(Json.newObject().set("health", Json.toJson(health))));
   }
 
+  public static Promise<Result> getExportPolicy(String datasetUrn) {
+    final DatasetExportPolicy exportPolicy;
+    try {
+      exportPolicy = EXPORT_POLICY_DAO.getDatasetExportPolicy(datasetUrn);
+    } catch (Exception e) {
+      if (e.toString().contains("Response status 404")) {
+        return Promise.promise(() -> notFound(_EMPTY_RESPONSE));
+      }
+
+      Logger.error("Fetch export policy fail", e);
+      return Promise.promise(() -> internalServerError(errorResponse(e)));
+    }
+
+    if (exportPolicy == null) {
+      return Promise.promise(() -> notFound(_EMPTY_RESPONSE));
+    }
+
+    return Promise.promise(() -> ok(Json.newObject().set("exportPolicy", Json.toJson(exportPolicy))));
+  }
+
+  public static Promise<Result> updateExportPolicy(String datasetUrn) {
+    final String username = session("user");
+    final DatasetExportPolicy exportPolicy;
```

Comment:
Can move this into "try"

Reply:
Won't that make exportPolicy only available in the scope inside the try block? Because I'm also trying to use it in the return statement below.

Reply:
Nevermind, I think comment below explains what you're going for